testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/singleton/partition/PartitionerServlet.java

```diff
@@ -36,25 +38,36 @@
 import org.jboss.as.server.CurrentServiceContainer;
 import org.jboss.msc.service.ServiceController;
 import org.jboss.msc.service.ServiceName;
+import org.jgroups.Address;
 import org.jgroups.Channel;
+import org.jgroups.Event;
 import org.jgroups.View;
 import org.jgroups.protocols.DISCARD;
 import org.jgroups.protocols.TP;
 import org.jgroups.protocols.pbcast.GMS;
 import org.jgroups.stack.ProtocolStack;
 
 /**
- * Servlet used to simulate network partitions. Responds to {@code /partitioner?partition=true} by inserting DISCARD protocol to the
- * stack and {@code /partitioner?partition=false} by removing previously inserted DISCARD protocol.
+ * Servlet used to simulate network partitions. Responds to {@code /partitioner?partition=true} by inserting DISCARD protocol to the stack
+ * and installing new view directly on the {@link GMS} and responds to {@code /partitioner?partition=false} by removing previously inserted
+ * {@link DISCARD} protocol and passing MERGE event up the stack.
+ * <p>
+ * Note that while it would be desirable for the tests to leave splitting and merging to the servers themselves, this is not practical in a
+ * test suite. While FD/VERIFY_SUSPECT/GMS can be be configured to detect partitions quickly the MERGE3 handles merges within randomized
+ * intervals and uses unreliable channel which can easily take several minutes for merge to actually happen. Also, speeds up the test
+ * significantly.
  *
  * @author Radoslav Husar
  */
 @WebServlet(urlPatterns = {PartitionerServlet.SERVLET_PATH})
 public class PartitionerServlet extends HttpServlet {
 
+    private static final long VIEWS_TIMEOUT = 3_000;
```

Comment:
TimeoutUtil.adjust()?

Reply:
This is running on server side and thus it has no knowledge of the passed in timeout property. Also, this is really not necessary.